src/eddymotion/data/dmri.py

```diff
@@ -40,34 +40,46 @@ def _data_repr(value):
     return f"<{'x'.join(str(v) for v in value.shape)} ({value.dtype})>"
 
 
+def _cmp(lh, rh):
+    if lh is None and rh is None:
+        return True
+    elif lh is None or rh is None:
+        return False
+    elif isinstance(lh, np.ndarray) and isinstance(rh, np.ndarray):
+        return np.allclose(lh, rh)
+    else:
+        raise NotImplementedError(f"Not implemented for {type(lh)} and {type(rh)}")
```

Comment:
May we relay on the original equality definitions of objects?

```suggestion
    if isinstance(lh, np.ndarray) and isinstance(rh, np.ndarray):
        return np.allclose(lh, rh)
    
    return lh == rh
```